ckanext/versions/blueprints.py

```diff
@@ -0,0 +1,45 @@
+from ckan import model
+from ckan.plugins import toolkit
+from flask import Blueprint
+
+from ckanext.versions.logic import action
+
+blueprint = Blueprint(
+    'versions',
+    __name__,
+)
+
+
+def version_download(id, resource_id, version):
+    """Download resource blueprint supporting version name.
+
+    This download blueprint gets the activity_id from the version and redirects
+    to a download url that can handle it. Currently is working with
+    blob_storage extension but can work with any download endpoint that knows
+    how to handle activities for resources.
+    """
+    context = {
+        'model': model,
+        'user': toolkit.c.user
+    }
+
+    try:
+        activity_id = action.get_activity_id_from_resource_version_name(
+            context, {'resource_id': resource_id, 'version_name': version})
```

Comment:
IIUC version names are arbitrary strings but not necessary URL friendly. Perhaps is better to use version['id'] in the URL? Or at least support both

Reply:
Great! I will use the id as you suggest for now. There is some work to improve the API design later on, I'm gonna do a work to accept both of them there.